src/components/contexts/AuthContext.tsx

```diff
@@ -0,0 +1,236 @@
+import { useMutation, useQuery, useQueryClient } from '@tanstack/react-query'
+import { apiClient } from 'api'
+import { authEndpoints, endpoints } from 'api/endpoints'
+import { InstitutionSelectModalProps } from 'components/organisms/modals/InstitutionSelectModal/InstitutionSelectModal'
+import { ModalTypes, showModal } from 'components/organisms/modals/ModalRoot'
+import useAsRef from 'hooks/useAsRef'
+import Cookies from 'js-cookie'
+import { size } from 'lodash'
+import {
+  FC,
+  PropsWithChildren,
+  createContext,
+  useCallback,
+  useContext,
+  useEffect,
+  useMemo,
+} from 'react'
+import { InstitutionDataType } from 'types/institutions'
+import { PrivilegeKey } from 'types/privileges'
+
+interface UserInfoType {
+  tolkevarav?: {
+    selectedInstitution?: {
+      id: string
+      name: string
+    }
+    surname?: string
+    forename?: string
+    privileges?: PrivilegeKey[]
+    institutionUserId?: string
+  }
+}
+
+interface AuthContextType {
+  initializing: boolean
+  isUserLoggedIn: boolean
+  login: () => void
+  logout: () => void
+  userInfo: UserInfoType
+  userPrivileges: PrivilegeKey[]
+  institutionUserId: string
+  institutions: InstitutionDataType[]
+  openInstitutionSelectModal: (
+    props: Partial<InstitutionSelectModalProps>
+  ) => void
+}
+
+const authContextDefaultValues: AuthContextType = {
+  initializing: true,
+  isUserLoggedIn: false,
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  login: () => {},
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  logout: () => {},
+  userInfo: {},
+  userPrivileges: [],
+  institutionUserId: '',
+  institutions: [],
+  // eslint-disable-next-line @typescript-eslint/no-empty-function
+  openInstitutionSelectModal: () => {},
+}
+
+const authContext = createContext<AuthContextType>(authContextDefaultValues)
+
+export const useAuth = () => {
+  return useContext(authContext)
+}
+
+export const rawLogin = () => {
+  window.location.href = `http://localhost:8000/login?redirect_uri=${window.location.href}`
+}
+
+export const rawLogout = (error?: string) => {
+  let redirectUri = window.location.origin
+
+  if (error) {
+    redirectUri += `#${error}`
+  }
+  window.location.href = `http://localhost:8000/logout?redirect_uri=${encodeURIComponent(
+    redirectUri
+  )}`
+}
+
+export const AuthProvider: FC<PropsWithChildren> = (props) => {
+  const { children } = props
+
+  const queryClient = useQueryClient()
+
+  const login = useCallback(() => {
```

Comment:
I don't think you need to wrap rawLogin inside useCallback

Reply:
True, but I wanted to provide similar API for other components using values from auth context. Since both methods are quite similar, then I wanted them to go through same layers: global definition -> context definition -> usage in context consumers